propagators/opentelemetry-propagator-aws-xray/src/AWSXRayPropagator.ts

```diff
@@ -92,9 +92,18 @@ export class AWSXRayPropagator implements TextMapPropagator {
     carrier: unknown,
     getter: TextMapGetter
   ): SpanContext {
-    const traceHeader = getter.get(carrier, AWSXRAY_TRACE_ID_HEADER);
-    if (!traceHeader || typeof traceHeader !== 'string')
+    const headerKeys = getter.keys(carrier);
+    const relevantHeaderKey = headerKeys.find((e) => {
+      return e.toLowerCase() === AWSXRAY_TRACE_ID_HEADER;
+    });
+    if (!relevantHeaderKey) {
       return INVALID_SPAN_CONTEXT;
+    }
+    const traceHeader = getter.get(carrier, relevantHeaderKey);
+
+    if (!traceHeader || typeof traceHeader !== 'string') {
+      return INVALID_SPAN_CONTEXT;
+    }
```

Comment:
Maybe it's better to implement that logic in a new proprietary TextMapGetter

Reply:
@osherv I don't think that's advisable.
If, as a platform engineer, I use the SDK to register an AWS XRay propagator, or a B3 propagator, that's my problem. It should be invisible to the API consumer.

The developer who then uses the API to extract or inject the context should only concern himself with the serialisation/deserialisation of the context into/from the payload. It should not be concerned whether or not the propagator is expecting a lowercase string or not.

If you stand your point, then the AWS Lambda instrumentation has to be modified to use case-independent header keys, because at the moment, using both appears to be incompatible.